File: 69.py
class MinStack:
    def __init__(self):
        self.stack = []
        self.minStack = []

    def push(self, x):
        self.stack.append(x)
        if not self.minStack or x <= self.minStack[-1]:
            self.minStack.append(x)
        else:
            self.minStack.append(self.minStack[-1])

    def pop(self):
        if self.stack:
            self.stack.pop()
            self.minStack.pop()

    def top(self):
        return self.stack[-1] if self.stack else None

    def getMin(self):
        return self.minStack[-1] if self.minStack else None


# Driver code
obj = MinStack()
obj.push(5)
obj.push(2)
obj.push(8)
print(obj.getMin())  # 2
obj.pop()
print(obj.top())     # 2
print(obj.getMin())  # 2

File: 69.js
class MinStack {
    constructor() {
        this.stack = [];
        this.minStack = [];
    }

    push(x) {
        this.stack.push(x);
        if (this.minStack.length === 0)
            this.minStack.push(x);
        else
            this.minStack.push(Math.min(x, this.minStack[this.minStack.length - 1]));
    }

    pop() {
        if (this.stack.length > 0) {
            this.stack.pop();
            this.minStack.pop();
        }
    }

    top() {
        return this.stack[this.stack.length - 1];
    }

    getMin() {
        return this.minStack[this.minStack.length - 1];
    }
}

// Driver Code
let obj = new MinStack();
obj.push(5);
obj.push(2);
obj.push(8);
console.log(obj.getMin()); // 2
obj.pop();
console.log(obj.top());    // 2
console.log(obj.getMin()); // 2
